hazelcast-client-legacy/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -401,9 +401,9 @@ protected Object putInternal(final K key, final V value, final ExpiryPolicy expi
             }
         } else {
             if (nearCache != null || statisticsEnabled) {
-                future.andThen(new ExecutionCallback<Object>() {
```

Comment:
@serkan-ozal @ahmetmircik 
That has been said, this solution although is doing all we want, seems complicated. 
Question I am trying to answer is that is trying to offload the work to user thread worth putting a complex code to system ?